Selectors/src/main/java/io/doist/recyclerviewext/choice_modes/Selector.java

```diff
@@ -50,16 +50,56 @@ public void toggleSelected(long id) {
 
     public abstract void clearSelected();
 
+    @Deprecated
     public void setOnSelectionChangedListener(@Nullable OnSelectionChangedListener observer) {
-        mObserver = observer;
+        if (observer != null) {
+            addOnSelectionChangedListener(observer);
+        } else {
+            removeAllUpdateListeners();
+        }
+    }
+
+    public void addOnSelectionChangedListener(@NonNull OnSelectionChangedListener observer) {
+        if (mObservers == null) {
+            mObservers = new ArrayList<>();
+        }
+        mObservers.add(observer);
+    }
+
+    public void removeOnSelectionChangedListener(@NonNull OnSelectionChangedListener observer) {
+        if (mObservers == null) {
+            return;
+        }
+        mObservers.remove(observer);
+        if (mObservers.size() == 0) {
+            mObservers = null;
+        }
+    }
+
+    public void removeAllUpdateListeners() {
```

Comment:
This name feels "out of place", when comparing to the others in this class (because of the `Update` part). 😅

I suggest we use something like `removeAllOnSelectionChangedListeners()`, `removeAllListeners()` or `clearListeners()`. 😊